aspnetcore/grpc/interceptors.md

```diff
@@ -0,0 +1,270 @@
+}
+```
+
+Overridden `AsyncUnaryCall` does the following:
+* Logs call to the console.
+* Adds caller metadata to call headers.
+* Handles response in a custom way (logs if an exception occurred).
+
+Although the interceptors for each kind of service method are slightly different, the concept behind `continuation` and `context` parameters remains the same.
+
+* `continuation` is a delegate which invokes the next interceptor in the chain or the underlying call invoker (if there is no interceptor left in the chain). It is not an error to call it zero or multiple times. Interceptors don't even have to return call representation (`AsyncUnaryCall` in case of unary RPC) returned from `continuation` delegate. Notice that in the preceding example a new instance of call representation is constructed. Omitting the delegate call and returning your own instance of call representation breaks the interceptors' chain and returns the associated response immediately.
+* `context` carries scoped-values associated with the client side call. You can use it to pass metadata like security principals, credentials or tracing data. Moreover, `context` carries information about deadlines and cancellation. For more information, see [Reliable gRPC services with deadlines and cancellation](xref:grpc/deadlines-cancellation#deadlines>).
+
+The following code shows how to add caller metadata through the context.
+
+```csharp
+private void AddCallerMetadata<TRequest, TResponse>(ref ClientInterceptorContext<TRequest, TResponse> context)
+    where TRequest : class
+    where TResponse : class
+{
+    var headers = context.Options.Headers;
+
+    // Call doesn't have a headers collection to add to.
+    // Need to create a new context with headers for the call.
+    if (headers == null)
+    {
+        headers = new Metadata();
+        var options = context.Options.WithHeaders(headers);
+        context = new ClientInterceptorContext<TRequest, TResponse>(context.Method, context.Host, options);
+    }
+
+    // Add caller metadata to call headers
+    headers.Add("caller-user", Environment.UserName);
+    headers.Add("caller-machine", Environment.MachineName);
+    headers.Add("caller-os", Environment.OSVersion.ToString());
+}
+```
+
+`LogCall`, `AddCallerMetadata` and `HandleResponse` methods are private methods embedded in the exemplary `ClientLoggerInterceptor` class. For complete implementation, see an [example of client interceptor](https://github.com/grpc/grpc-dotnet/blob/master/examples/Interceptor/Client/ClientLoggerInterceptor.cs).
+
+### Configure client interceptors
+
+gRPC client interceptors are configured by creating a call invoker from a channel and then 
+creating a client from that invoker.
+
+The following code:
+* Creates a channel by using `GrpcChannel.ForAddress`.
+* Creates an invoker by using `Intercept` method on the created channel.
+* Creates a client from the invoker.
```

Comment:
```suggestion
gRPC client interceptors are configured on a channel.

The following code:
* Creates a channel by using `GrpcChannel.ForAddress`.
* Uses the `Intercept(...)` extension method to configure the channel to use the interceptor. Note that this method returns a `CallInvoker`. Strongly typed gRPC clients may be created from an invoker just like a channel.
* Creates a client from the invoker. gRPC calls made by the client automatically execute the interceptor.
```
IMO because a client can be created from a channel and call invoker in the same way there isn't a need to stress their differences.